From 7a6c0e84831e9d61d4549267a32de1beb7133ffc Mon Sep 17 00:00:00 2001
From: Carlos Pargada <carlos.pargada@intel.com>
Date: Fri, 15 Feb 2013 17:41:03 +0100
Subject: [PORT FROM MAIN]use memcpy as memmove

BZ: 84683

In some apps, memcpy usage can overwrite memory.
This fix solve issue but lost code optimisation

Change-Id: I6df10716c2745698c952430dca019d5ad2b2e3e3
Orig-Change-Id: Ib7e3b64b0193eef039975d039dcb6702efd1d2f3
Category: device enablement
Domain: AOSP-Bionic
Origin: internal
Upstream-Candidate: no
Signed-off-by: Carlos Pargada <carlos.pargada@intel.com>
---
 libc/arch-x86/string/ssse3-memcpy-atom.S |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/libc/arch-x86/string/ssse3-memcpy-atom.S b/libc/arch-x86/string/ssse3-memcpy-atom.S
index 1080a38..5013886 100644
--- a/libc/arch-x86/string/ssse3-memcpy-atom.S
+++ b/libc/arch-x86/string/ssse3-memcpy-atom.S
@@ -31,6 +31,10 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "cache.h"
 #undef __i686
 
+#ifndef USE_AS_MEMMOVE
+# define USE_AS_MEMMOVE 1
+#endif
+
 #ifndef MEMCPY
 # define MEMCPY	memcpy
 #endif
-- 
1.7.4.1

